### DISCUSSION TITLE
S3 PutBucketPolicy Response

### DISCUSSION THREAD

**Discussion by Nugine** [awslabs/aws-sdk-rust#729] [Q&A] [answered]
The [documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_control_PutBucketPolicy.html#API_control_PutBucketPolicy_ResponseSyntax) and [model](https://github.com/awslabs/aws-sdk-rust/blob/66423e05991ee831696bc32fe3e452694cf0d231/aws-models/s3.json) say that PutBucketPolicy returns 200 OK. But the [sample response](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketPolicy.html#API_PutBucketPolicy_Example_2) is 204 No Content. 
Which one is correct? The client may accept both. But the server must choose one.

Related:
+ [minio/mc rejects 200](https://github.com/minio/mc/issues/2393#issuecomment-367447570)
+ [aws-sdk-s3 rejects 204](https://github.com/awslabs/aws-sdk-rust/blob/66423e05991ee831696bc32fe3e452694cf0d231/sdk/s3/src/operation.rs#L2886-L2898)
+ [minio/mint expects 204](https://github.com/minio/mint/blob/225cf1a37944504d4c11309519ecb97edd87e30a/run/core/aws-sdk-php/quick-tests.php#L790-L792)
+ https://github.com/Nugine/s3s/issues/4

**Comment by rcoh** [comment, marked as answer]
The docs and the model are the canonical source, as is the Rust SDK (assuming it works!) It's possible it changed at some point in history?

You may also want to investigate the behavior of other AWS SDKs (e.g. Java / Python)
